src/hooks/useExtraSafePaddingBottomStyle.ts

```diff
@@ -0,0 +1,34 @@
+import {useEffect, useMemo, useState} from 'react';
+import {Keyboard} from 'react-native';
+import useStyledSafeAreaInsets from './useStyledSafeAreaInsets';
+import useThemeStyles from './useThemeStyles';
+
+const useExtraSafePaddingBottomStyle = () => {
+    const styles = useThemeStyles();
+    const [willKeyboardShow, setWillKeyboardShow] = useState(false);
```

Comment:
I added the type for it. But I don't think we should rename the state, because as I mentioned above, ```willKeyboardShow``` is used to fix the issue ```didKeyboardShow``` encountered: There is a delay when the keyboard is visible by the user and the when ```didKeyboardShow``` is set to ```true```. So it is not just ```isKeyboardVisible```.